src/main/scala/LSWidgets.scala

```diff
@@ -52,73 +54,40 @@ class ProcedureWidget(val key: WidgetKey, val state: State, val ws: GUIWorkspace
     },
     false
   )
-  val springLayout = new SpringLayout()
-  setLayout(springLayout)
-  val bigSpace = 5
-  val smallSpace = 3
-
-  val saveButton = new JButton("save")
-  saveButton.onActionPerformed(_ =>
-    try ws.evaluateCommands(owner, saveCommand, ws.world.observers, false)
-    catch { case e: CompilerException => ws.warningMessage(e.getMessage) }
-  )
-  springLayout.putConstraint(WEST, saveButton, bigSpace, WEST, this)
-  springLayout.putConstraint(NORTH, saveButton, bigSpace, NORTH, this)
-  add(saveButton)
 
-  val deleteButton = new JButton("delete")
-  deleteButton.onActionPerformed(_ =>
-    try ws.evaluateCommands(owner, deleteCommand, ws.world.observers, waitForCompletion = false)
-    catch { case e: CompilerException => ws.warningMessage(e.getMessage) }
-  )
-  springLayout.putConstraint(NORTH, deleteButton, bigSpace, NORTH, this)
-  springLayout.putConstraint(WEST, deleteButton, bigSpace, EAST, saveButton)
-  add(deleteButton)
-
-  val typeChooser: XWComboBox = new XWComboBox(() => updateInState(kind.typeProperty))
-  typeChooser.items = Seq("agentset", "command", "reporter")
-  springLayout.putConstraint(WEST, typeChooser, bigSpace, EAST, deleteButton)
-  springLayout.putConstraint(NORTH, typeChooser, bigSpace, NORTH, this)
-  add(typeChooser)
-
-  typeChooser.onItemStateChanged { event ⇒
-    if (event.getStateChange == ItemEvent.SELECTED)
-      updateInState(kind.typeProperty)
+  val saveButton: JButton = makeButton(
+    "agentset.save", tryCompilation(ws, owner, () => saveCommand))
+  val deleteButton: JButton = makeButton(
+    "agentset.delete", tryCompilation(ws, owner, () => deleteCommand))
+  val nameField: JTextField = boundTextField(10, kind.nameProperty)
+  val argField: JTextField = boundTextField(0, kind.argProperty)
+
+  removeAll()
+  setLayout(new BoxLayout(this, BoxLayout.Y_AXIS))
+  add(buttonPanel)
+  add(Box.createRigidArea(new Dimension(0, 3)))
+  add(inputPanel)
+
+  def buttonPanel: JPanel = {
```

Comment:
Shouldn't this be a `val`, not a `def`? Right now, calling `buttonPanel` multiple times will create a new panel every time.